components/net/image_cache.rs

```diff
@@ -85,28 +91,8 @@ fn set_webrender_image_key(webrender_api: &webrender_api::RenderApi, image: &mut
     image.id = Some(image_key);
 }
 
-// TODO(gw): This is a port of the old is_image_opaque code from WR.
-//           Consider using SIMD to speed this up if it shows in profiles.
-fn is_image_opaque(format: webrender_api::ImageFormat, bytes: &[u8]) -> bool {
-    match format {
-        webrender_api::ImageFormat::BGRA8 => {
-            let mut is_opaque = true;
-            for i in 0..(bytes.len() / 4) {
-                if bytes[i * 4 + 3] != 255 {
-                    is_opaque = false;
-                    break;
-                }
-            }
-            is_opaque
-        }
-        webrender_api::ImageFormat::RGB8 => true,
-        webrender_api::ImageFormat::RG8 => true,
-        webrender_api::ImageFormat::A8 => false,
-        webrender_api::ImageFormat::Invalid | webrender_api::ImageFormat::RGBAF32 => unreachable!(),
-    }
-}
-
-fn premultiply(data: &mut [u8]) {
+fn premultiply(data: &mut [u8]) -> bool {
```

Comment:
Maybe add docs for the return value here? It's obvious if you read the source but...